index.js

```diff
@@ -6,7 +6,7 @@ module.exports = function (userCb) {
 		this.paths.push(file.path);
 
 		if (userCb) {
-			userCb(file.path, function () {
+			userCb(file.path).then(function () {
```

Comment:
There's potential here for an unhandled promise rejection here. Is a rejected promise from the user cause enough to end the stream? Maybe we could warn and continue?